### PR TITLE
Use bidirectional streaming RPCs for scan operations in DistributedStorageService

### PR DESCRIPTION
**Pull request by brfrn169**
This PR changes the scan operations for `DistributedStorageService` to use bidirectional streaming RPCs in gRPC. I basically did the same things in #228. 

Please take a look!

### CODE REVIEW

**Comment by brfrn169** [comment]
I rebased this to fix the conflicts. Please take a look!

File: rpc/src/main/proto/scalardb.proto [hunk 1]
```diff
@@ -122,11 +122,7 @@ message Result {
 service DistributedStorage {
   rpc Get (GetRequest) returns (GetResponse) {
   }
-  rpc OpenScanner (OpenScannerRequest) returns (OpenScannerResponse) {
-  }
-  rpc ScanNext (ScanNextRequest) returns (ScanNextResponse) {
-  }
-  rpc CloseScanner (CloseScannerRequest) returns (google.protobuf.Empty) {
+  rpc Scan (stream ScanRequest) returns (stream ScanResponse) {
```
Comment (feeblefakie) [inline rpc/src/main/proto/scalardb.proto:125]: Nice!

File: rpc/src/main/proto/scalardb.proto [hunk 2]
```diff
@@ -140,31 +136,16 @@ message GetResponse {
   Result result = 1;
 }
 
-message OpenScannerRequest {
+message ScanRequest {
   Scan scan = 1;
-  int32 fetch_count = 2;
+  optional int32 fetch_count = 2;
```
Comment (feeblefakie) [inline rpc/src/main/proto/scalardb.proto:141]: Oh, I didn't know `optional` is back now 👍

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**